calliope/backend/pyomo/util.py

```diff
@@ -217,5 +217,5 @@ def check_sign(var):
         return "Any"
 
 
-def check_value(val: po.base.param._ParamData) -> bool:
-    return val._value == po.base.param._NotValid or po.value(val) is None
+def is_valid(val: po.base.param._ParamData) -> bool:
+    return not (val._value == po.base.param._NotValid or po.value(val) is None)
```

Comment:
I think renaming `is_valid` to something like `invalid` and reversing the logic would be simpler in terms of readability.  The comparison itself is to `_NotValid`, which is negated, and in `pyomo/constraints/group.py` it is negated again with `not is_valid(...)`. And in `pyomo/constraints/costs.py`, the checks remain very readable: `loc_tech_is_in(..) and not invalid(..)`.

PS: Regarding readability, my mental model for readability for conditionals is "Does this read like a natural sentence?"  On occasion you can also apply this to loops and simplify them.

Reply:
makes a lot of sense, I went with it (after doing some cleanup to deal with all the failing tests).